src/main/webapp/components/Security.js

```diff
@@ -0,0 +1,278 @@
+    };
+  }
+
+  componentDidCatch(error, info) {
+    console.log('----------------------- error ----------------------')
+    console.log(error, info);
+  }
+
+  static getDerivedStateFromError(error) {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true }
+  }
+
+  handleRadio2Change = (e, field, value) => {
+    this.setState(prev => {
+      prev.formData[field] = value;
+      return prev;
+    }, () => {
+      this.props.updateForm(this.state.formData, field);
+      this.props.handleSecurityValidity(this.validate());
+      this.props.removeErrorMessage();
+    });
+  };
+
+  handleInputChange = (e) => {
+    const field = e.target.name;
+    const value = e.target.value;
+    this.setState(prev => {
+      prev.formData[field] = value;
+      return prev;
+    }, () => {
+      this.props.handleSecurityValidity(this.validate());
+      this.props.updateForm(this.state.formData, field);
+      this.props.removeErrorMessage();
+    })
+  };
+
+  validate = (field) => {
+    let pii = false;
+    let compliance = false;
+    let sensitive = false;
+    let accessible = false;
+    let isValid = true;
+    let textCompliance = false;
+    let textSensitive = false;
+    let textAccessible = false;
+
+    if (isEmpty(this.state.formData.pii)) {
+      pii = true;
+      isValid = false;
+    }
+    if (isEmpty(this.state.formData.compliance)) {
+      compliance = true;
+      isValid = false;
+    }
+
+    if (!isEmpty(this.state.formData.compliance)
+        && this.state.formData.compliance === "true"
+        && isEmpty(this.state.formData.textCompliance)) {
+      textCompliance = true;
+      isValid = false;
+    }
+    if (isEmpty(this.state.formData.sensitive)) {
+      sensitive = true;
+      isValid = false;
+    }
+    if (!isEmpty(this.state.formData.sensitive) && this.state.formData.sensitive === "true" && isEmpty(this.state.formData.textSensitive)) {
+      textSensitive = true;
+      isValid = false;
+    }
+    if (isEmpty(this.state.formData.accessible)) {
+      accessible = true;
+      isValid = false;
+    }
+    if (!isEmpty(this.state.formData.accessible) && this.state.formData.accessible === "true" && isEmpty(this.state.formData.textAccessible)) {
+      textAccessible = true;
+      isValid = false;
+    }
+    if (field === undefined || field === null || field === 3) {
+      this.setState(prev => {
+        prev.errors.pii = pii;
+        prev.errors.compliance = compliance;
+        prev.errors.sensitive = sensitive;
+        prev.errors.accessible = accessible;
+        prev.errors.textCompliance = textCompliance;
+        prev.errors.textSensitive = textSensitive;
+        prev.errors.textAccessible = textAccessible;
+        return prev;
+      });
+    }
+    else if (field === 'pii' || field === 'compliance' || field === 'textCompliance' || field === 'sensitive'
+      || field === 'textSensitive' || field === 'accessible' || field === 'textAccessible') {
```

Comment:
These field comparisons (here and below) seem very brittle to me. It looks like there's a 1 to 1 alignment with both the `state.formData` and `state.errors` properties. The comparisons are also very repetitive. It would be nice if we could, instead, say something like this (warning, pseudo-code):

```
else if (fieldNames.contains(field)) {
  this.setState(prev => {
    fieldNames.each { f -> prev.errors.get(f) = f }
    return prev
});
```
That would save a whole lot of boilerplate code here and below. I don't have a great idea about how to do this in javascript, but maybe if `state.formData` and `state.errors` were generalized into an object, that might make it easier.

Reply:
I hope I have not been too drastic on this. 
I the midle of doing this change I saw it was not neccesary anymore to check by each specific field. This was used on the update functions above, but I think it is unnecessary now. 

If you see the entire validate function, you will see it first validates all fields in the form, to later after go to this portion of  the code, where it would only update that specific error flag in the state... functionality will not change  if we just update all flags in the state at once, without caring individual fields, and we save logic and code in the process.
Does it make sense for you?

Reply:
It does make sense, thank you!